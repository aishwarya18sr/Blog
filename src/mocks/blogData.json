[
  {
    "id": 1,
    "date": "2023-01-15T23:30:00.001Z",
    "readingTime": "2 mins",
    "title": "The future of abstract art and the culture ...",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 10,
    "liked": false,
    "image": "abstract.png"
  },
  {
    "id": 2,
    "date": "2023-01-16T23:30:00.001Z",
    "readingTime": "1 min",
    "title": "How painting outdoors can transform your ",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 1,
    "liked": true,
    "image": "outdoor.png"
  },
  {
    "id": 3,
    "date": "2023-01-14T23:30:00.001Z",
    "readingTime": "2 mins",
    "title": "10 young painters you need to know",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 15,
    "liked": false,
    "image": "young-painters.png"
  },
  {
    "id": 4,
    "date": "2023-01-17T23:30:00.001Z",
    "readingTime": "5 mins",
    "title": "Tips for creating the perfect art",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 11,
    "liked": false,
    "image": "perfect-art.png"
  },
  {
    "id": 5,
    "date": "2023-01-18T23:30:00.001Z",
    "readingTime": "9 mins",
    "title": "Stay inspired: top artists blog ...",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 45,
    "liked": true,
    "image": "inspired.png"
  },
  {
    "id": 6,
    "date": "2023-01-20T23:30:00.001Z",
    "readingTime": "2 mins",
    "title": "5 reasons why everyone should try an...",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 11,
    "liked": false,
    "image": "reason.png"
  }
]
