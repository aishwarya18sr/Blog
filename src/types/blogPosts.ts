export interface BlogData {
  id: number;
  image: string;
  date: string;
  readingTime: string;
  title: string;
  description: string;
  claps: number;
  liked: boolean;
}

export interface ClapsData {
  data: {
    claps: 10;
    liked: false;
  };
}
