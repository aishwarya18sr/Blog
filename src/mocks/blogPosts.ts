import { BlogData } from "../types";

export const mockBlogPostData: BlogData[] = [
  {
    id: 1,
    date: "6/29/2021 4:52:48 PM UTC",
    readingTime: "2 mins",
    title: "mock title 1",
    description: "mock description 1",
    claps: 10,
    liked: false,
    image: "abstract.png",
  },
  {
    id: 2,
    date: "6/29/2022 4:52:48 PM UTC",
    readingTime: "4 mins",
    title: "mock title 2",
    description: "mock description 2",
    claps: 12,
    liked: true,
    image: "abstract.png",
  },
];

export const mockClapsData = {
  data: {
    claps: 10,
    liked: false,
  },
};
