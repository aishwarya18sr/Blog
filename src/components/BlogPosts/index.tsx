import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import { GET_BLOG_DATA } from "../../constants/apiEndPoints";
import { BlogData } from "../../types";
import { getBlogIndexById } from "../../utils/common";
import makeRequest from "../../utils/makeRequest";
import BlogPostCard from "../BlogPostCard";
import "./blogPosts.css";

const BlogPosts = () => {
  const [allBlogData, setAllBlogData] = useState<BlogData[]>([]);
  const navigate = useNavigate();

  useEffect(() => {
    makeRequest(GET_BLOG_DATA, {}, navigate).then((response) => {
      setAllBlogData(response);
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return allBlogData ? (
    <div className="posts basic-padding">
      {allBlogData.map((eachBlogData) => {
        return (
          <BlogPostCard
            key={eachBlogData.id}
            blogData={eachBlogData}
            setAllBlogData={(updatedBlogData: BlogData) => {
              const blogDataIndex = getBlogIndexById(
                allBlogData,
                updatedBlogData.id
              );
              setAllBlogData([
                ...allBlogData.slice(0, blogDataIndex),
                updatedBlogData,
                ...allBlogData.slice(blogDataIndex + 1),
              ]);
            }}
          />
        );
      })}
    </div>
  ) : (
    <div className="blogDataLoader">
      <p>Loading...</p>
    </div>
  );
};

export default BlogPosts;
