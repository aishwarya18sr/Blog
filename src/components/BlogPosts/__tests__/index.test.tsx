import {
  act,
  render,
  renderHook,
  screen,
  waitFor,
} from "@testing-library/react";
import React from "react";
import { mockBlogPostData } from "../../../mocks/blogPosts";
import BlogPosts from "..";

jest.mock("../../../utils/makeReuqest/", () => () => {
  return Promise.resolve(mockBlogPostData);
});

describe("BlogPosts", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(<BlogPosts />);
    await waitFor(() => {
      expect(screen.getByText("mock title 1")).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
