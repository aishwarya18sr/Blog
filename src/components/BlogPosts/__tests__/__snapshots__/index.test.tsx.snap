// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPosts should render correctly 1`] = `
<DocumentFragment>
  <div
    class="posts basic-padding"
  >
    <div
      class="post"
    >
      <div
        class="post-image"
      >
        <img
          alt=""
          src="abstract.png"
        />
      </div>
      <div
        class="post-content post-padding"
      >
        <div
          class="post-meta"
        >
          <div
            class="post-date"
          >
            29th
      June, 
      2021
          </div>
          <div
            class="post-reading-time"
          >
            2 mins
          </div>
        </div>
        <div
          class="post-title"
        >
          mock title 1
        </div>
        <div
          class="post-description"
        >
          mock description 1
        </div>
      </div>
      <div
        class="post-engagements post-padding"
      >
        <div
          class="clap"
        >
          <div
            class="clap-icon"
          >
            <img
              alt="clap"
              src="clapping.svg"
            />
          </div>
          <div
            class="clap-counter"
          >
            10
          </div>
        </div>
        <div
          class="like"
        >
          <img
            alt="heart"
            src="heart-black.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="post"
    >
      <div
        class="post-image"
      >
        <img
          alt=""
          src="abstract.png"
        />
      </div>
      <div
        class="post-content post-padding"
      >
        <div
          class="post-meta"
        >
          <div
            class="post-date"
          >
            29th
      June, 
      2022
          </div>
          <div
            class="post-reading-time"
          >
            4 mins
          </div>
        </div>
        <div
          class="post-title"
        >
          mock title 2
        </div>
        <div
          class="post-description"
        >
          mock description 2
        </div>
      </div>
      <div
        class="post-engagements post-padding"
      >
        <div
          class="clap"
        >
          <div
            class="clap-icon"
          >
            <img
              alt="clap"
              src="clapping.svg"
            />
          </div>
          <div
            class="clap-counter"
          >
            12
          </div>
        </div>
        <div
          class="like"
        >
          <img
            alt="heart"
            src="heart-red.svg"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
